modules/core/nuxeo-platform-commandline-executor/src/main/java/org/nuxeo/ecm/platform/commandline/executor/service/EnvironmentDescriptor.java

```diff
@@ -80,4 +74,27 @@ public String getName() {
         return name;
     }
 
+    /**
+     * Returns a new descriptor, merging a global environment with another one.
+     *
+     * @since 11.5
+     */
+    public static EnvironmentDescriptor newInstance(EnvironmentDescriptor global, EnvironmentDescriptor other) {
+        EnvironmentDescriptor res = new EnvironmentDescriptor();
+        if (global != null) {
+            if (global.workingDirectory != null) {
+                res.workingDirectory = global.workingDirectory;
+            }
+            res.parameters.putAll(global.getParameters());
+        }
+        if (other != null) {
+            res.name = other.name;
+            if (other.workingDirectory != null) {
+                res.workingDirectory = other.workingDirectory;
+            }
+            res.parameters.putAll(other.getParameters());
+        }
+        return res;
+    }
```

Comment:
I ended up doing this because i removed the "merge" method (that was impacting existing descriptor instances), I'll have a look at what you suggest but note there was no unit test, so I hope this will not break the expected behaviour.

Reply:
I added back the old merge method